### PR TITLE
Py keras extension

### PR DESCRIPTION
**Pull request by bvanessen**
Added support for GPUs in Theano and added the py-pexpect package for ipython.

### CODE REVIEW

File: var/spack/repos/builtin/packages/libgpuarray/package.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    homepage = "http://deeplearning.net/software/libgpuarray/"
+    url      = "https://github.com/Theano/libgpuarray/archive/v0.6.1.tar.gz"
+
+    version('0.6.2', '7f163bd5f48f399cd6e608ee3d528ee4')
+    version('0.6.1', 'cfcd1b54447f9d55b05514df62c70ae2')
+    version('0.6.0', '98a4ec1b4c8f225f0b89c18b899a000b')
+
+    depends_on('cuda')
+    depends_on('python')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libgpuarray/package.py]: `extends('python')` implies `depends_on('python')` so you don't need this line.

**Comment by bvanessen** [comment]
Please note that this set of packages encounters a problem once it is installed. When I try to load it with spack activate py-keras actually generates an error related to py-theano already being activated. It seems like the system is trying to activate theano twice and doesn't recognize that it is already active. I started talking with Greg about this problem. @becker33 @tgamblin. If I change the python package for Keras to only have a build dependency then this problem goes aways, but that is not correct, it should have the run dependency.

File: var/spack/repos/builtin/packages/libgpuarray/package.py (hunk 1 above)
Comment (bvanessen) [inline var/spack/repos/builtin/packages/libgpuarray/package.py]: Okay, removed that dependency.

**Comment by adamjstewart** [comment]
@bvanessen Are you saying that from a clean Spack clone, if you run:
```
$ spack install py-keras
$ spack activate py-keras
```
it complains that `py-theano` is already activated? Or perhaps did you run:
```
$ spack install py-theano
$ spack activate py-theano
$ spack install py-keras
$ spack activate py-keras
```
In that case it should correctly mention that `py-theano` is already activated. This isn't an error message, just a helpful warning.

**Comment by bvanessen** [comment]
See inline comments:

Brian C. Van Essen
vanessen1@llnl.gov
(w) 925-422-9300
(c) 925-290-5470





> On Mar 16, 2017, at 1:20 PM, Adam J. Stewart <notifications@github.com> wrote:
> 
> @bvanessen Are you saying that from a clean Spack clone, if you run:
> 
> $ spack install py-keras
> $ spack activate py-keras
> 
> it complains that py-theano is already activated?

Yes, when I try to activate py-keras without having already activated py-theano, it will fail and say that py-theano is already activated.


> Or perhaps did you run:

No, I wasn’t trying to do this.

**Pull request merged**